server/src/main/java/io/crate/execution/engine/aggregation/impl/TopKAggregation.java

```diff
@@ -436,13 +470,41 @@ int id() {
         }
     }
 
+    private static boolean supportedByLongSketch(DataType<?> type) {
+        return switch (type.id()) {
+            case LongType.ID -> true;
+            case DoubleType.ID -> true;
+            case FloatType.ID -> true;
```

Comment:
Given that we support float which is 4 bytes and just expand int to long, can't we do the same for Integer type? 
We wouldn't even need NumericUtils, just `(long) int_val` in `toLong` and (int) long_val in `toObject`

In addition to ^: looks like we can also express all numeric types via `long` and integral types (byte, short) can be handled by up/down casting